packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -157,17 +147,23 @@ export const createTransactionsTracker = (
     )
   }: TransactionsTrackerInternals = {}
 ): TransactionsTracker => {
-  const directionalTransactions$ = createDirectionalTransactionsTrackerSubject(transactionsSource$, addresses$);
-  const providerTransactionsByDirection$ = (direction: TransactionDirection) =>
-    directionalTransactions$.pipe(
-      map((transactions) => transactions.filter((tx) => tx.direction === direction).map(({ tx }) => tx)),
+  const historicalTransactions$ = createHistoricalTransactionsTrackerSubject(transactionsSource$);
+  const providerTransactionsByDirection$ = (inspector: SendReceiveValueInspector) =>
+    combineLatest([historicalTransactions$, addresses$]).pipe(
+      map(([transactions, addresses]) => {
+        const inspectTx = createTxInspector({
+          value: inspector(addresses)
+        });
+        return transactions.filter((tx) => inspectTx(tx).value.coins > 0n);
+      }),
       distinctUntilChanged(transactionsEquals)
     );
+
   const incomingTransactionHistory$ = new TrackerSubject<Cardano.TxAlonzo[]>(
-    providerTransactionsByDirection$(TransactionDirection.Incoming)
+    providerTransactionsByDirection$(valueReceivedInspector)
```

Comment:
This will classify transactions that have any change outputs as incoming, which is not what we want here. This observable should emit transactions where no coin was sent from the wallet